chain/chain/src/resharding/manager.rs

```diff
@@ -0,0 +1,141 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use near_chain_configs::{MutableConfigValue, ReshardingConfig, ReshardingHandle};
+use near_chain_primitives::Error;
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::block::Block;
+use near_primitives::challenge::PartialState;
+use near_primitives::hash::CryptoHash;
+use near_primitives::shard_layout::get_block_shard_uid;
+use near_primitives::stateless_validation::stored_chunk_state_transition_data::StoredChunkStateTransitionData;
+use near_primitives::types::chunk_extra::ChunkExtra;
+use near_primitives::types::AccountId;
+use near_primitives::utils::get_block_shard_id;
+use near_store::trie::mem::resharding::RetainMode;
+use near_store::{DBCol, PartialStorage, ShardTries, ShardUId, Store};
+
+pub struct ReshardingManager {
+    store: Store,
+    epoch_manager: Arc<dyn EpochManagerAdapter>,
+    /// Configuration for resharding.
+    pub resharding_config: MutableConfigValue<ReshardingConfig>,
```

Comment:
Not for this PR; The old config won't make much sense for Instant Resharding. We can reuse the same config for it or introduce a separate config. I think we need to keep the old fields for now as we still use it for the archival nodes recovery.